### PR TITLE
fix: Deallocate row addresses and size arrays after exporting

### PR DESCRIPTION
**Pull request by viirya**
## Which issue does this PR close?

Closes #.

## Rationale for this change

When we export row addresses and sizes to native, they are copied to separate arrays. We should deallocate array buffers after exporting them to reduce memory allocation.

### CODE REVIEW

**Comment by viirya** [comment]
cc @sunchao @parthchandra

**Review by sunchao — APPROVED** [review]

File: spark/src/main/scala/org/apache/spark/shuffle/sort/RowPartition.scala [hunk 1]
```diff
@@ -32,8 +32,17 @@ class RowPartition(initialSize: Int) {
...
+    val array = rowAddresses.toArray
+    rowAddresses = null
+    array
+  }
+
+  def getRowSizes: Array[Int] = {
+    val array = rowSizes.toArray
+    rowSizes = null
+    array
+  }
```
[the remark this answers is not in the document]
Comment (viirya) [reply spark/src/main/scala/org/apache/spark/shuffle/sort/RowPartition.scala:45]: They won't be accessed again after `getRowAddresses` and `getRowSizes` get called. Once they are called, they are exported to native side for spilling. After that, `RowPartition` will be reset.

[the remark this answers is not in the document]
Comment (viirya) [reply spark/src/main/scala/org/apache/spark/shuffle/sort/RowPartition.scala]: i.e.,

```
long addrs[] = getRowAddresses();
long sizes[] = getRowSizes;
nativeSpilling(adds, sizes);
rowPartition.reset();
```

**Comment by viirya** [comment]
Merged. Thanks.

**Pull request merged**